File: 1395-minimum-time-visiting-all-points/minimum-time-visiting-all-points.py
class Solution:
    def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
        time = 0
        x1 , y1 = points.pop()     #we start from the last element because its the same thing both ways
        while points != []:        #couldve done while points:
            x2,y2 = points.pop()
            time += max(abs(x1-x2) , abs(y1-y2))
            x1,y1 = x2,y2
        return time

File: 1395-minimum-time-visiting-all-points/minimum-time-visiting-all-points.java
class Solution {
    public int minTimeToVisitAllPoints(int[][] points) {
        int count = 0;
        int[] cursor = Arrays.copyOf(points[0], 2);
        for (int i = 1; i < points.length; i++) {
            while (cursor[0] != points[i][0] || cursor[1] != points[i][1]) {
                if (cursor[0] < points[i][0]) cursor[0]++;
                else if (cursor[0] > points[i][0]) cursor[0]--;

                if (cursor[1] < points[i][1]) cursor[1]++;
                else if (cursor[1] > points[i][1]) cursor[1]--;

                count++;
            }
        }
        return count;
    }
}